src/core/gnsspositioninformation.h

```diff
@@ -116,19 +116,19 @@ class GnssPositionInformation
      * Altitude (in meters) above or below the mean sea level.
      */
     double elevation() const { return mElevation; }
-    bool elevationValid() const { return mElevation != 0; }
+    bool elevationValid() const { return !std::isnan( mElevation ); }
```

Comment:
I agree.
But I don't understand something: Why we initialize the member variables in `GnssPositionInformatio` (I did it analogous to `QgsGpsInformation`) some values with `0` and others with `quiet_NaN()`. Does it make a difference? Should we make it for all the same?

Reply:
@signedav , I think it'd be good, can we do this in another PR, I'm keen to merge this one. We also need to update the latitudeValid check to not do a != 0 check (who knows, in fifty years, maybe null island will actually be a physical settlement, we want QField to be future-proof :wink: